src/actions/TodoActions.js

```diff
@@ -0,0 +1,196 @@
+import { API } from '../api/tasksAPI';
+import {
+	LOAD_DATA,
+	ADD_TODOS,
+	CHANGE_TYPE,
+	CHANGE_TYPE_ALL,
+	DELETE_TASK,
+	EDIT_TASK,
+	CLEAR_TASK_COMPLETED,
+} from '../constants/nameTypes';
+import { calledServer, callingServer } from './LoadingActions';
+
+const loadData = (value) => {
+	return {
+		type: LOAD_DATA,
+		payload: value,
+	};
+};
+
+const addTodos = (value) => {
+	return {
+		type: ADD_TODOS,
+		payload: value,
+	};
+};
+
+const changeType = (value) => {
+	return {
+		type: CHANGE_TYPE,
+		payload: value,
+	};
+};
+
+const changeTypeAll = (value) => {
+	return {
+		type: CHANGE_TYPE_ALL,
+		payload: value,
+	};
+};
+
+const deleteTask = (value) => {
+	return {
+		type: DELETE_TASK,
+		payload: value,
+	};
+};
+
+const editTask = (value) => {
+	console.log(value);
+	return {
+		type: EDIT_TASK,
+		payload: {
+			id: value.id,
+			content: value.value.content,
+		},
+	};
+};
+
+const clearTaskCompleted = () => {
+	return {
+		type: CLEAR_TASK_COMPLETED,
+	};
+};
+
+// middleware
+
+const loadDB = (param) => {
+	return (dispatch) => {
+		dispatch(callingServer());
+		API.callListTask(param)
+			.then((res) => {
+				dispatch(loadData(res.data));
+				dispatch(calledServer());
+			})
+			.catch((err) => {
+				console.log(err);
+				dispatch(calledServer());
+			});
+	};
+};
+
+const createTask = (value) => {
+	return (dispatch) => {
+		dispatch(callingServer());
+		API.createNewTask(value)
+			.then((res) => {
+				dispatch(addTodos(res.data));
+				dispatch(calledServer());
+			})
+			.catch((err) => {
+				console.log(err);
+				dispatch(calledServer());
+			});
+	};
+};
+
+const isFinishAll = (value) => {
```

Comment:
Name meaning? isFinishAll mean true or fail but function not return true or fail

Reply:
Updated!